### PR TITLE
[#2623] Robustness bad variables

### LINKED ISSUE #2623: Improve robustness/visibility of "bad" variables used in choice-widgets
Context: https://github.com/open-formulieren/open-forms/pull/2595#discussion_r1089188861

#2595 adds the feature to populate the dropdown/radio/selectboxes values from a variable. This _will_ be combined in the future with the feature to fetch data from remote services and store (a subset/subpath) of them in variables.

Currently the implementation has a line to check whether the json logic expression needs to be applied or not:

```py
is_obj = isinstance(items_array[0], dict)
```

This is fragile if the `items_array` is not homogenous - there's no guarantee that all items are a dict or even have the same keys/structure (response data from other APIs may be sparse, this is especially prevalent with the existing StUF services!).

What happens (expected) then is that the `jsonLogic` call for the lookup will fallback to the `None` value, possibly yielding non-unique values.

**Tasks**

- [ ] Add a check for duplicate values - log these both to the submission via `logevent` and to Sentry via `logger.warning(...)`
- [ ] If there are duplicate values, make them unique by adding a suffix (`0`, `1` etc.) (?)

### PR DESCRIPTION
**Pull request by SilviaAmAm**
Fixes #2623

### CODE REVIEW

**Review by CharString — CHANGES_REQUESTED** [review]
I think we should assert the logevent if we know we trigger it.

And on a more bird's eye view of the problem we're trying to solve here... Is `valueExpression` even needed? As far as I can see it can be expressed with `map` in the `itemExpression` or since that is missing in our jsonlogic fork: a `reduce` of the `var` with a `merge` on an empty array as the accumulator.

**Comment by CharString** [comment]
I don't know if this is brand new functionality or if `valueExpression` is in use in the wild.
If new, I'd rather implement a simple `map` and reduce (no pun intended) complexity in this code... Having `map` while strictly not needed, does make expressing cherry picking values off a repeating group *that* much more ergonomic.

And I can see its use in triggers too for step triggers like "Do you have children under the age of 18 in your household." Those would just be a "any" of a "map" of a comparison of birthdates in the repeating group of co-inhabitants. Steps like this exist in tax forms.

**Comment by SilviaAmAm** [comment]
It is new functionality from the previous PR! https://github.com/open-formulieren/open-forms/pull/2595

I could add the `map` to the JSON logic package and then we could get rid of `valueExpression`!

File: src/openforms/formio/dynamic_config/tests/test_dynamic_options.py [hunk 1]
```diff
@@ -562,3 +562,114 @@ def test_wrong_type_variable(self):
...
+                    "i'm not an object!",
+                    123,
+                    ["im", "an", "array"],
+                ]
+            },
+        )
+        self.assertEqual(
+            configuration["components"][0]["values"],
+            [{"label": "111", "value": "111"}],
+        )
```
[the remark this answers is not in the document]
Comment (SilviaAmAm) [reply src/openforms/formio/dynamic_config/tests/test_dynamic_options.py:675]: I don't think so: atm the log event is triggered if there is no valueExpression and the items are not primitives :thinking:

**Comment by CharString** [comment]
Great! The most beautiful code, is the  code that doesn't exist... :fire: > :sparkles: in my opinion. :wink:

**Comment by SilviaAmAm** [comment]
Still needs upgrading json-logic-py: https://github.com/maykinmedia/json-logic-py/pull/15

**Review by sergei-maertens — CHANGES_REQUESTED** [review]
Had a quick discussion with Chris about this because I was concerned this makes it impossible for us to specify different sources for the values and labels of choices - as we now basically always normalize to a list of strings and every item (string) is used as both value and label.

However, that approach makes it impossible to query other systems based on "system data" (like IDs) and still displaying the human readable label. It should be possible to use a variable with the contents:

```js
[
  {dbId: 1, name: 'Foo'},
  {dbId: 2, name: 'Bar'},
]
```
as the data source for select/checkboxes/radio.

This should ideally result in:

```
<select>
  <option value="1">Foo</option>
  <option value="2">Bar</option>
</select>
```

The same can go for a repeating group with children for example, where you enter the BSN and name as separate fields and use the BSN as value while displaying the name.

Long story short - the map expression should return either a list of strings, which we normalize to a list of two-tuples (value and label are the same then), or a list of two-tuples representing `(value, label)`.

Our gut-feeling is that you can come up with a `map` expression that achieves this, but we'd like to see this confirmed in the implementation and tests.

**Review by CharString — APPROVED** [review]
No real issues. This is fine as is. :+1:
